scene/gui/text_edit.cpp

```diff
@@ -2340,6 +2340,9 @@ void TextEdit::_move_caret_left(bool p_select, bool p_move_by_word) {
 				set_caret_line(get_caret_line(i) - 1, false, true, -1, i);
 				set_caret_column(text[get_caret_line(i)].length(), i == 0, i);
 			} else {
+				if (is_use_custom_separators_enabled()) { //&& get_word_separators().size() != 0) {
+					TS->shaped_text_set_custom_punctuation(text.get_line_data(get_caret_line(i))->get_rid(), get_word_separators());
```

Comment:
This should be set when line RIDs are initialized, if it's used here, it will do unnecessary full text reprocessing and won't apply custom separators to the line breaking.

Reply:
Any suggestions where it may best fit?

Reply:
In the `TextEdit::Text::invalidate_cache` and `TextEdit::Text::invalidate_all_lines()` (see how `brk_flags` are handled, for example).

- Add `set_word_separators` implementation to the `TextEdit::Text`.
- Add `TS->shaped_text_set_custom_punctuation` call to the `Text::invalidate_*`.
- Call `text.set_word_separators` and `text.invalidate_all_lines` from the `set_use_custom_separators`.

Reply:
Great tip thanks! However, not I caused a bug I guess:

1) If we change the enable/disable custom separators flag
2)  change file
3) close using X without saving the file
4) click on save & quit option

does not save the file.

If I do not touch the status of the custom separators flag and perform 2 3 4, it does save the file... Any help how I caused this?

Reply:
Apparently this bug was already there. I will create an issue on godot for this.